### PR TITLE
Add inheritance to orpha

### LINKED ISSUE #4398: Update command that downloads ORPHA data to include inheritance file
So far if you run `scout download orpha` you just download file orphadata_en_product4.xml and orphadata_en_product6.xml, but you need also file 9 to include inheritance data in orpha terms.

### PR DESCRIPTION
**Pull request by TereseBo**
This PR adds inheritance for Orphacodes by including the relevant orphadata product in the download of orpha files, parsing of the file and updating the orpha disease_terms to include the information. 

Fix #4398

The tests for parsing of orpha files and disease_terms have been updated accordingly.

A reduced copy of the downloaded file has also been included in the resource files for use in the demo setup and tests.

<details>
<summary>Testing on cg-vm1 server (Clinical Genomics Stockholm)</summary>

**Prepare for testing**
1. Make sure the PR is pushed and available on [Docker Hub](https://hub.docker.com/repository/docker/clinicalgenomics/scout-server-stage)
1. Fist book your testing time using the Pax software available at [https://pax.scilifelab.se/](https://pax.scilifelab.se). The resource you are going to call dibs on is `scout-stage` and the server is `cg-vm1`.
1. `ssh <USER.NAME>@cg-vm1.scilifelab.se`
1. `sudo -iu hiseq.clinical`
1. `ssh localhost`
1. (optional) Find out which scout branch is currently deployed on cg-vm1: `podman ps`
1. Stop the service with current deployed branch: `systemctl --user stop scout.target`
1. Start the scout service with the branch to test: `systemctl --user start scout@<this_branch>`
1. Make sure the branch is deployed: `systemctl --user status scout.target`
1. After testing is done, repeat procedure at [https://pax.scilifelab.se/](https://pax.scilifelab.se), which will release the allocated resource (`scout-stage`) to be used for testing by other users.
</details>

<details>
<summary>Testing on hasta server (Clinical Genomics Stockholm)</summary>

**Prepare for testing**
1. `ssh <USER.NAME>@hasta.scilifelab.se`
1. Book your testing time using the Pax software. `us; paxa -u <user> -s hasta -r scout-stage`. You can also use the WSGI Pax app available at [https://pax.scilifelab.se/](https://pax.scilifelab.se).
1. (optional) Find out which scout branch is currently deployed on cg-vm1: `conda activate S_scout; pip freeze | grep scout-browser`
1. Deploy the branch to test: `bash /home/proj/production/servers/resources/hasta.scilifelab.se/update-tool-stage.sh -e S_scout -t scout -b <this_branch>`
1. Make sure the branch is deployed: `us; scout --version`
1. After testing is done, repeat the `paxa` procedure, which will release the allocated resource (`scout-stage`) to be used for testing by other users.
</details>

**How to test**:
1. how to test it, possibly with real cases/data

**Expected outcome**:
The functionality should be working
Take a screenshot and attach or copy/paste the output.

**Review:**
- [x] code approved by CR
- [x] tests executed by TB

### CODE REVIEW

**Comment by TereseBo** [comment]
Download command tested on hasta:
Screenshot 2024-01-29 at 14 46 19

Screenshot 2024-01-29 at 14 47 49

After updating the diseases, inheritance information is also present for orphaterms:
Screenshot 2024-01-29 at 14 56 15

Screenshot 2024-01-29 at 14 54 53

File: scout/parse/disease_terms.py [hunk 1]
```diff
@@ -115,23 +125,52 @@ def get_omim_disease_terms(genemap_lines: List = None, hpo_annotation_lines: Lis
...
+
     #: Combine information
     orpha_disease_terms: Dict = consolidate_gene_and_hpo_annotation(
         gene_annotations=orpha_disease, hpo_annotations=orpha_hpo_annotations
     )
+
     return orpha_disease_terms
+
+
+def add_inheritance_information(orpha_disease, orpha_inheritance):
```
Comment (northwestwitch) [inline scout/parse/disease_terms.py:162]: Type hints and return type

File: scout/parse/omim.py [hunk 2]
```diff
@@ -83,9 +65,9 @@ def parse_genemap2_phenotypes(phenotype_entry, mim_number=None):
         # Find the inheritance
         inheritance = set()
         inheritance_text = ",".join(splitted_info[i:])
-        for term in MIM_INHERITANCE_TERMS:
+        for term in DISEASE_INHERITANCE_TERMS:
```
Comment (northwestwitch) [inline scout/parse/omim.py:68]: 💯

File: scout/parse/omim.py [hunk 3]
```diff
@@ -1,32 +1,14 @@
...
-    "Autosomal recessive",
-    "Autosomal dominant",
-    "Digenic recessive",
-    "X-linked dominant",
-    "X-linked recessive",
-    "Y-linked",
-    "Mitochondrial",
-]
-
-TERMS_MAPPER = {
```
Comment (northwestwitch) [inline scout/parse/omim.py:19]: Great to reuse for all disease types

**Review by northwestwitch — APPROVED** [review]
Great addition! Works fine and looks good code-wise! 🥇

**Pull request merged**